### PR TITLE
feat: allow MAX_VALIDATOR_WITHDRAWAL_REQUESTS_PER_TRANSACTION=0 to disable withdrawals

### LINKED ISSUE #2480: Allow MAX_VALIDATOR_WITHDRAWAL_REQUESTS_PER_TRANSACTION=0 to disable withdrawals
## Description

Allow `MAX_VALIDATOR_WITHDRAWAL_REQUESTS_PER_TRANSACTION=0` to disable all withdrawal submissions (partial withdrawals and validator exits) while the rest of the automation service continues running normally (monitoring, metrics, yield reporting).

## Motivation

Setting the env var to `0` is the simplest way to disable all withdrawal submissions, without stopping the rest of the service

## Tasks

- [x] Change schema validation from `positive()` to `nonnegative()` for `MAX_VALIDATOR_WITHDRAWAL_REQUESTS_PER_TRANSACTION`
- [x] Add schema tests for zero acceptance and negative rejection
- [x] Add regression tests in BeaconChainStakingClient for `maxValidatorsPerTx=0`

## Acceptance criteria

- [x] `MAX_VALIDATOR_WITHDRAWAL_REQUESTS_PER_TRANSACTION=0` parses successfully
- [x] Negative values are still rejected
- [x] With value 0, no `unstake` calls are made for partial withdrawals or validator exits
- [x] All existing tests continue to pass

## Risks

- [ ] None significant - the loop logic already handles 0 correctly (breaks immediately), so the only change is relaxing the schema validation

### PR DESCRIPTION
**Pull request by kyzooghost**
## Summary

Closes #2480

- Change `MAX_VALIDATOR_WITHDRAWAL_REQUESTS_PER_TRANSACTION` schema validation from `positive()` to `nonnegative()` so setting it to `0` disables all withdrawal submissions (partial withdrawals and validator exits)
- Add schema tests for zero acceptance and negative rejection
- Add regression tests confirming `unstake` is never called when `maxValidatorsPerTx=0`

**Context:** QuickNode's Hoodi beacon node returns stale `pending_partial_withdrawals` data, causing duplicate withdrawal requests. Setting the value to `0` is the simplest kill switch while the RPC issue is resolved - the rest of the service (monitoring, metrics, yield reporting) continues normally.

## Test plan

- [x] Schema test: `"0"` parses to `0` successfully
- [x] Schema test: `"-1"` is rejected
- [x] Client test: `submitWithdrawalRequestsToFulfilAmount` with `maxValidatorsPerTx=0` never calls `unstake`
- [x] Client test: `submitMaxAvailableWithdrawalRequests` with `maxValidatorsPerTx=0` never calls `unstake` (covers both partial withdrawals and exits)
- [x] All 49 schema tests pass, all 18 client tests pass

---

> [!NOTE]
> **Low Risk**
> Small validation change guarded by tests; only affects behavior when this env var is explicitly set to `0`.
> 
> **Overview**
> Allows `MAX_VALIDATOR_WITHDRAWAL_REQUESTS_PER_TRANSACTION` to be set to `0` as an operational kill-switch, disabling submission of both partial withdrawal requests and validator exits while still rejecting negative values.
> 
> Adds/updates unit tests to confirm the config schema parses `"0"` to `0` and that `BeaconChainStakingClient` never calls `unstake` when configured with `maxValidatorsPerTx=0` (for both `submitWithdrawalRequestsToFulfilAmount` and `submitMaxAvailableWithdrawalRequests`).
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit 148fb1a595641de60e77e49c7632a8202f106cbb. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

### CODE REVIEW

File: native-yield-operations/automation-service/src/application/main/config/config.schema.ts [hunk 1]
```diff
@@ -108,7 +108,8 @@ export const configSchema = z
       .transform((val) => BigInt(val))
       .refine((v) => v >= 0n, { message: "Must be nonnegative" }),
     // Maximum number of validator withdrawal requests that will be batched in a single transaction.
-    MAX_VALIDATOR_WITHDRAWAL_REQUESTS_PER_TRANSACTION: z.coerce.number().int().positive(),
+    // Set to 0 to disable all withdrawal submissions (partial withdrawals and validator exits).
+    MAX_VALIDATOR_WITHDRAWAL_REQUESTS_PER_TRANSACTION: z.coerce.number().int().nonnegative(),
```
Comment (thedarkjester) [inline native-yield-operations/automation-service/src/application/main/config/config.schema.ts:112]: where is this consumed? I see it being added, but never used

Comment (kyzooghost) [reply native-yield-operations/automation-service/src/application/main/config/config.schema.ts:112]: config is preexisting

It gets mapped to `maxValidatorWithdrawalRequestsPerTransaction` in config.ts
-> Used in BeaconChainStakingClient to cap the number of validator withdrawals (both partial and full)
